nums/numpy/api.py

```diff
@@ -444,6 +446,29 @@ def any(a: BlockArray, axis=None, out=None, keepdims=False):
     return _instance().reduce("any", a, axis=axis, keepdims=keepdims)
 
 
+def average(a: BlockArray,
+            axis: Union[None, int] = None,
+            weights: Union[None, BlockArray] = None,
+            returned: bool = False):
+    if weights is None:
+        avg = mean(a, axis)
+        weights_sum = BlockArray.from_scalar(a.size / avg.size, a.system)
+    else:
+        if a.shape != weights.shape:
+            raise NotImplementedError("'average' currently does not support broadcasting;"
+                                      "dimensions of 'a' and 'weights' must match.")
+        weights_sum = sum(weights, axis=axis)
+        if not all(weights_sum):
```

Comment:
Just want to check: Is this what NumPy raises if there are zeros in some entries of `weights_sum`? Can you test that this error is raised when there are zeros?

Reply:
Yes, NumPy source code checks if `any` element is equal to zero. Will add tests to show this.